model/plotWidget.cc

```diff
@@ -427,7 +427,7 @@ namespace minsky
     // determine if any of the incoming vectors has a ptime-based xVector
     xIsSecsSinceEpoch=false;
     for (auto& i: yvars)
-      if (i && xvars[&i-&yvars[0]] && !i->hypercube().xvectors.empty())
+      if (i && xvars[&i-yvars.data()] && !i->hypercube().xvectors.empty())
```

Comment:
This is probably mostly stylistic, but I suppose &x[0] does involve adding 0 to the pointer, so I suppose data() is better.

Reply:
It is not merely stylistic and it is not the reason why I did it. The problem lies in that it might trigger an assertion in the stdlib in case the vector is empty, because it may try accessing the element and then return the pointer.

Reply:
Assertion will not be triggered, as the code is not run when yvars is empty.